.github/workflows/build.yml

```diff
@@ -1,61 +1,62 @@
-name: "Build"
+name: Build
 
 on:
   - push
   - pull_request
 
 jobs:
-  build: # Check build on various OSes
-
+  build:
     strategy:
+      fail-fast: false
       matrix:
         os:
           - ubuntu-latest
+          - macos-latest
+          - windows-latest
         ocaml-compiler:
-          - 4.02.x
-          - 4.04.x
-          - 4.06.x
-          - 4.08.x
-          - 4.10.x
-          - 4.12.x
-          - 4.14.x
-          - 5.0.x
-          - 5.1.x
+          - 4
+          - 5
         include:
           - os: ubuntu-latest
-            ocaml-compiler: 5.1.x
+            # We don't need to run lower bound test for more than one build
+            ocaml-compiler: "4.02"
+          - os: ubuntu-latest
+            ocaml-compiler: 5
             # We don't need to compute coverage for more than one build
             send-coverage: true
             # Mdx tests Mdx tests
             run-mdx: true
-      fail-fast: false
 
     runs-on: ${{ matrix.os }}
 
     steps:
-      # Clone the project
-      - uses: actions/checkout@v2
+      - name: Set git to use LF
+        if: runner.os == 'Windows'
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+
+      - name: Checkout tree
+        uses: actions/checkout@v4
 
-      # Setup
-      - name: Setup OCaml ${{ matrix.ocaml-version }}
-        uses: ocaml/setup-ocaml@v2
+      - name: Setup OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v3
         with:
-          # Do not pin odoc to not break Mdx installation
-          opam-pin: false
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
-          opam-local-packages: |
-            odoc.opam
+          # Do not pin odoc to not break Mdx installation
+          opam-pin: ${{ matrix.run-mdx != true }}
 
       - name: Install dependencies
-        run: opam install -y --deps-only -t ./odoc.opam ./odoc-parser.opam
+        run: opam install --deps-only --with-test odoc odoc-parser
```

Comment:
Won't that install deps from the odoc from opam repository in case where `run-mdx` is true, breaking if we modify the dependencies without releasing?

Reply:
Indeed.

Reply:
Fixed